src/main/webapp/javascript/login.js

```diff
@@ -0,0 +1,104 @@
+function hideError() {
+    var errorPanel = document.getElementById("error-panel");
+    errorPanel.classList.add("w3-hide");
+}
+
+function showError(msg) {
+    var errorPanel = document.getElementById("error-panel");
+    errorPanel.classList.remove("w3-hide");
+    w3DisplayData("error-panel", {"message": msg});
+}
+
+function switchRegistration() {
+    hideError();
+    var checkbox = document.getElementById("register-cb");
+    var pwd2 = document.getElementById("password2-grp");
+    var loginBtn = document.getElementById("login-btn");
+    var registerBtn = document.getElementById("register-btn");
+    if (checkbox.checked) {
+        pwd2.classList.remove("w3-hide");
+        loginBtn.classList.add("w3-hide");
+        registerBtn.classList.remove("w3-hide");
+    } else {
+        pwd2.classList.add("w3-hide");
+        loginBtn.classList.remove("w3-hide");
+        registerBtn.classList.add("w3-hide");
+    }
+}
+
+function login() {
+    hideError();
+    console.log("start login");
+    var usernameTxt = document.getElementById("username-txt");
+    var passwordTxt = document.getElementById("password1-txt");
+    var dto = {
+        "username": usernameTxt.value,
+        "password": passwordTxt.value
+    };
+    console.log("sending login data");
+    fetch('/api/auth/login', {
+        "method": "POST",
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(dto)
+    }).then(function (response) {
+        if (response.status === 200) {
+            console.log("login success");
+            location.href = '/app/start.jsp';
```

Comment:
This should be extracted into a function in the future - like in `addEvents()`, `main()`, etc. in `event_logic.js`.